src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -520,6 +520,33 @@ internal static SecurityKey ResolveTokenSigningKey(string kid, string x5t, Token
             return null;
         }
 
+        /// <summary>
+        /// Counts the number of Jwt Token segments.
+        /// </summary>
+        /// <param name="token">The Jwt Token.</param>
+        /// <param name="maxCount">The maximum number of segments to count up to.</param>
+        /// <returns>The number of segments up to <paramref name="maxCount"/>.</returns>
+        internal static int CountJwtTokenPart(string token, int maxCount)
```

Comment:
Note, on `net8.0`, this method could take advantage of the new `MemoryExtensions.Count` method, which can take advantage of SIMD operations to make the count even faster.

Reply:
will we do that in this PR w/some Ifdefs?

Reply:
We would need to target `net8.0`, which we aren't doing:

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/ddd4a4c01d2c05cb65323c5bb49e81579f2322b3/build/targets.props#L3

It would also involve using the 8.0 SDK to build. So I would say we should do that in a different PR.